Mathlib/LinearAlgebra/TensorProduct/Basis.lean

```diff
@@ -66,5 +66,81 @@ lemma Basis.baseChange_apply (b : Basis ι R M) (i) :
     b.baseChange S i = 1 ⊗ₜ b i := by
   simp [Basis.baseChange, Basis.tensorProduct]
 
+section
+
+variable (ℬ : Basis ι R M) (𝒞 : Basis κ R N) (x : M ⊗[R] N)
+
+/--
+If `{𝒞ᵢ}` is a basis for the module `N`, then every elements of `x ∈ M ⊗ N` can be uniquely written
+as `∑ᵢ mᵢ ⊗ 𝒞ᵢ` for some `mᵢ ∈ M`.
+-/
+def TensorProduct.equivFinsuppOfBasisRight : M ⊗[R] N ≃ₗ[R] κ →₀ M :=
```

Comment:
I would have gone for
```lean
def TensorProduct.equivFinsuppOfBasisRight : M ⊗[R] N ≃ₗ[R] κ →₀ M := by
  classical -- for DecidableEq κ
  exact LinearEquiv.lTensor M 𝒞.repr ≪≫ₗ TensorProduct.finsuppScalarRight R M κ
```

instead of your 25 lines. But this breaks the next proof, annoyingly. A fix is
```lean
lemma TensorProduct.equivFinsuppOfBasisRight_symm_apply (b : κ →₀ M) :
    (TensorProduct.equivFinsuppOfBasisRight 𝒞).symm b = b.sum fun i m ↦ m ⊗ₜ 𝒞 i := by
  classical
  induction b using Finsupp.induction_linear
  · simp
  · simp_all [Finsupp.sum_add_index, add_tmul]
  · simp [equivFinsuppOfBasisRight]
```

I don't have a good feeling about why `rfl` fails. Ultimately do you think that this approach is a win?

Reply:
as long as `TensorProduct.equivFinsuppOfBasisRight_symm_apply` works, all the proofs are very easy to fix. So I have adapted to your definition

instead of `classical -- for DecidableEq κ`, I just assumed `DecidableEq \kappa`